JsonFlatFileDataStore/DataStore.cs

```diff
@@ -44,9 +44,9 @@ public DataStore(string path, bool useLowerCamelCase = true, string keyProperty
                             // Serializing JObject ignores SerializerSettings, so we have to first deserialize to ExpandoObject and then serialize
                             // http://json.codeplex.com/workitem/23853
                             var jObject = JsonConvert.DeserializeObject<ExpandoObject>(data.ToString());
-                            return JsonConvert.SerializeObject(jObject, Formatting.Indented, _serializerSettings);
+                            return JsonConvert.SerializeObject(jObject, minifyJson ? Formatting.None: Formatting.Indented, _serializerSettings);
```

Comment:
Selected formatting could be done before setting `_toJsonFunc`.

```
var usedFormatting = minifyJson ? Formatting.None: Formatting.Indented;
```

Then there is no need to have conditional check every time function is executed.